File: pythonCode/contrastAdjustmentHistogramEqualization.py
import cv2
import numpy as np

#read image
image = cv2.imread("../assets/putin.jpg")

#convert to YCrCb color space
imageYcb = cv2.cvtColor(image, cv2.COLOR_BGR2YCrCb)

#split into channels
Y, C, B = cv2.split(imageYcb)

#histogram equalization
Y = cv2.equalizeHist(Y)

#merge the channels
imageYcb = cv2.merge([Y, C, B])

#convert back to BGR color space
result = cv2.cvtColor(imageYcb, cv2.COLOR_YCrCb2BGR)

#create windows to display image
cv2.namedWindow("image", cv2.WINDOW_NORMAL)
cv2.namedWindow("result", cv2.WINDOW_NORMAL)

#display images
cv2.imshow("image", image)
cv2.imshow("result", result)

#press Esc to exit the program
cv2.waitKey(0)

#close all the opended windows
cv2.destroyAllWindows()

File: cPlusPlusCode/contrastAdjustmentHistogramEqualization.cpp
#include<opencv2/imgproc.hpp>
#include<opencv2/highgui.hpp>
#include<iostream>

using namespace cv;
using namespace std;

int main(){

    //Read image
    Mat image = imread("../assets/putin.jpg");

    //convert to YcrCb color space
    Mat imageYcb;
    cvtColor(image, imageYcb, COLOR_BGR2YCrCb);

    //split into channels
    vector<Mat> channels(3);
    split(imageYcb, channels);

    //histogram equalizarion of Y channel
    equalizeHist(channels[0], channels[0]);

    //merge the channels
    merge(channels, imageYcb);

    //convert back to BGR color space
    Mat result;
    cvtColor(imageYcb, result, COLOR_YCrCb2BGR);

    //create windows to display images
    namedWindow("image", WINDOW_NORMAL);
    namedWindow("result", WINDOW_NORMAL);

    //display images
    imshow("image", image);
    imshow("result", result);

    //press esc to exit the program
    waitKey(0);

    //close all the opended windows
    return 0;
}